src/peergos/server/tests/simulation/Simulator.java

```diff
@@ -287,8 +299,10 @@ private void write(String user, Path path) {
         Path dirName = path.getParent();
         String fileName = path.getFileName().toString();
         Map<Path, List<String>> dirsToFiles = index.getDirToFiles(user);
-        dirsToFiles.get(dirName).add(fileName);
-
+        List<String> existingFiles = dirsToFiles.get(dirName);
+        if (! existingFiles.contains(fileName)) {
```

Comment:
this  makes me a bit nervous - should `existingFiles` be updated to a set instead?

Reply:
done